service/src/main/java/uk/nhs/adaptors/gpc/consumer/utils/FindAPatientDocsUtil.java

```diff
@@ -0,0 +1,61 @@
+package uk.nhs.adaptors.gpc.consumer.utils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gpc.consumer.filters.FindAPatientDocsGatewayFilterFactory;
+
+@UtilityClass
+@Slf4j
+public class FindAPatientDocsUtil {
+    public static String replaceUrl(FindAPatientDocsGatewayFilterFactory.Config config, String body) {
+        LOGGER.info(String.format("Replace host: %s, to: %s", config.getGpcUrl(), config.getGpcConsumerUrl()));
+        return body.replace(config.getGpcUrl(), config.getGpcConsumerUrl());
+    }
+
+    public ByteArrayOutputStream zipStringToOutputStream(String responseWithProxyUrlReplacement) throws Exception {
+        try {
+            ByteArrayOutputStream obj = new ByteArrayOutputStream();
+            GZIPOutputStream gzip = new GZIPOutputStream(obj);
+            gzip.write(responseWithProxyUrlReplacement.getBytes(UTF_8));
+            gzip.close();
+
+            LOGGER.info("Response body successfully compress");
+
+            return obj;
+        } catch (Exception e) {
+            LOGGER.error("Error occuring compressing response: %s", e);
+            throw new Exception("Error occuring compressing response", e);
```

Comment:
```suggestion
            LOGGER.error("Error occurring compressing response: %s", e);
            throw new Exception("Error occurring compressing response", e);
```

Reply:
You need to show me how you make these suggestions haha they're brilliant